### ISSUE TITLE
about the "Base_Dataset"

### ISSUE THREAD

**Issue by liujianzhao6328057** [BUserName/PGL#3]
Hi, thanks for your interesting work. I have some doubt about the Base_Dataset. 
First, in "generate_new_train_data", a new known and unknown images list are generated. Assume that the length of new_label_flag is N1. The length of all target images is N2. In the Base_Dataset, the length of self.target_image is N2 and the length of self.label_flag is N1. The two list cannot correspond to each other.  Why 
            if self.partition == 'train':
                if self.target_ratio > 0:
                    index = random.choice(list(range(len(self.label_flag))))
                else:
                    index = random.choice(list(range(len(self.target_image))))
                # index = random.choice(list(range(len(self.label_flag))))
                target_image = Image.open(self.target_image[index]).convert('RGB')
Second, in the paper, the pseudo labelled target images are combined step by step. However, in the "generate_new_train_data", the  new_label_flag is newly generated at each step and is not combined with the previously selected pseudo images list. 

Expect your reply and best wishes.

**Comment by BUserName** [comment]
Hi jianzhao,

Thanks for your interest in our work! Currently, the server is running out of resources for me to re-run the code. I will see if I can found a chance today or early tmr. 

Thanks again for your patience,
Zijian

**Comment by BUserName** [comment]
Hi Jianzhao, 

I have investigated the questions.  The slight difference between the number of 'target_image' and 'label_flag' is related to the last batch in the dataloader. Yet, in the label estimation stage, the order of label_flag and target_image remains the same, so that it still makes the correct selection. 

In regarding the second question, we mark the unselected pos / unk data in the function 'select_top_data'.

Thanks again for pointing out this potential issue, we will update the data loader in the near future! You can also reach me at zijian.wang at uq.edu.au, if you have any further questions.
